src/xdp-utils.c

```diff
@@ -929,6 +929,9 @@ xdp_get_alternate_document_path (const char *path, const char *app_id)
 {
   int len;
 
+  if (g_str_equal (app_id, ""))
```

Comment:
`g_str_equal()` is **not** NULL-safe, is it possible that `app_id` is NULL here? If so, it'd be more appropriate to use `app_id == NULL || g_str_equal(app_id, "")`, or perhaps `g_strcmp0()`

Reply:
No, `xdp_app_info_get_id` is guaranteed to return a non-NULL value. I have some docs written to document that guarantee that I'm going to push with the follow-up PR.